File: front-end/src/app/service-models/app/v1/client/company-accounts-transaction-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';
import { ExpenseModeSM } from '../../enums/expense-mode-s-m.enum';

export class CompanyAccountsTransactionSM extends CoinManagementServiceModelBase<number> {
    expenseName!: string;
    expensePurpose!: string;
    expenseAmount!: number;
    expenseDate!: Date;
    currencyCode!: string;
    expenseMode!: ExpenseModeSM;
    expensePaid!: boolean;
    clientCompanyDetailId!: number;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/CompanyAccountsTransactionSM.cs
using Siffrum.Web.Payroll.ServiceModels.Base;
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class CompanyAccountsTransactionSM : SiffrumPayrollServiceModelBase<int>
    {
        public string ExpenseName { get; set; }
        public string ExpensePurpose { get; set; }
        public float ExpenseAmount { get; set; }
        public DateTime ExpenseDate { get; set; }
        public string CurrencyCode { get; set; }
        public ExpenseModeSM ExpenseMode { get; set; }
        public bool ExpensePaid { get; set; }
        public int ClientCompanyDetailId { get; set; }
    }
}
